### ISSUE TITLE
Fine-tuning Multi-Band MelGAN yields pure noise as soon as discriminator starts training

### ISSUE THREAD

**Issue by ZDisket** [dathudeptrai/TensorflowTTS#33]
As the title says, I started fine-tuning the published mb-melgan v1 model and set the discriminator to start at 10k steps. As soon as it started, all the[ sample audios](https://drive.google.com/drive/folders/1DMkLKRLQ7Find-XtZwh9Chzlm2u5BsAU) became pure noise.
I can confirm it's not the dataset since I could train MelGAN-STFT. If it helps, I'm using mixed precision for the generator but not the discriminator. I also had this problem with kan-bayashi's mb-melgan v2.

**Comment by dathudeptrai** [comment]
did u train G 200k steps ?

**Comment by ZDisket** [comment]
@dathudeptrai The audio sounded pretty good except for metallic noises at 10k, so I didn't train the generator for 200k, just 10k. And here's the [tensorboard](https://a97c91b97548.ngrok.io/#scalars).
I'll try messing with the config.

**Comment by dathudeptrai** [comment]
@ZDisket There is something wrong, loss too high, can you try training from scratch ?. Or can u try just load weight for generator ?. Loss jump from 0.5x to 600, very weird

**Comment by dathudeptrai** [comment]
In case training from scratch not work, you can share me thr dataset, i will try to fix

**Comment by ZDisket** [comment]
@dathudeptrai I'll try training again only loading generator.

**Comment by ZDisket** [comment]
Loading generator only causes a different kind of noise. The only thing remaining is to train from scratch.
[15000steps.zip](https://github.com/dathudeptrai/TensorflowTTS/files/4743773/15000steps.zip)

**Comment by dathudeptrai** [comment]
@ZDisket hmm let try, maybe the discriminator too strong even it's just start to train ?. In this case maybe you can try replace kernel_size=downsample_scale * 10 + 1 to kernel_size=downsample_scale * smaler_number + 1

**Comment by ZDisket** [comment]
@dathudeptrai Where do I change kernel_size? I don't see it in the training python file.

**Comment by dathudeptrai** [comment]
https://github.com/dathudeptrai/TensorflowTTS/blob/master/tensorflow_tts/models/melgan.py#L335 @ZDisket. I don't know what is the problem, but after u training from scratch if it's still bug, send me the dataset then i will try.

**Comment by ZDisket** [comment]
@dathudeptrai Okay, I'll use `kernel_size=downsample_scale * 2 + 1` with fine tuning if training from scratch doesn't work.

**Comment by ZDisket** [comment]
@dathudeptrai 
I can confirm training from scratch with `kernel_size=downsample_scale * 2 + 1` yields the same results. I uploaded the [predictions](https://ufile.io/p0onqatj). 10k or less is before discriminator.
I can send you the dataset and notebook I use to train by email.

**Comment by dathudeptrai** [comment]
@ZDisket that ok.

**Comment by ZDisket** [comment]
@dathudeptrai You closed the issue. I assume you found a solution?

**Comment by ZDisket** [comment]
@dathudeptrai Yes, MelGAN-STFT works well. I have my [notebook ](https://colab.research.google.com/drive/1SLl0tVT6tBu1zEn7atlrMWMIk1ucLRd0?usp=sharing)where you can use the model I'm training. I'll give you the dataset and explain more by email.
